docs/3.1-Shadow-Config.md

```diff
@@ -106,6 +106,14 @@ Type: "error" OR "warning" OR "info" OR "debug" OR "trace"
 
 Log level of output written on stdout. If Shadow was built in release mode, then messages at level 'trace' will always be dropped.
 
+#### `general.parallelism`
+
+Default: null  
+Type: Integer OR null
+
+How many parallel threads to use to run the simulation. Optimal performance is
+usually obtained with `nproc`, or sometimes `nproc/2` with hyperthreading.
```

Comment:
Do we need to explain what happens if set to null? Or maybe this isn't the place for that?

Reply:
I don't think there was a way for the user to explicitly set this to NULL. This reminded me though that there was some hidden logic that 0 would be silently treated as 1, and that `worker-threads=0` is special.

I cleaned this up by changing `parallel` to be of type `NonZeroU32`, and added the documentation about `worker-threads=0`.